### DISCUSSION TITLE
Creating a font-family utility with abitrary values does not work as expected

### DISCUSSION THREAD

**Discussion by vaelu** [tailwindlabs/tailwindcss#12940] [Help] [answered]
Hi folks. 👋

I'm trying to create font-family utilities with css variables in arbitrary values, like this:

```css
:root {
  --headline1-font-family: "Open Sans", sans-serif;
}
```
```html
<h1 class="font-[--headline1-font-family]">Title</h1>
```

Since Tailwind does of course not know what the value of the css variable is, it creates a font-weight utility, instead of font-family.

According to [this docs](https://tailwindcss.com/docs/adding-custom-styles#resolving-ambiguities), it should be possible to pass along the type of the css variable, which tells Tailwind what utility it should generate. I assume that font-sizes are strings (see [here](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Types)), so I tried `font-[string:--headline1-font-family]`, which unfortunately does not work.

What is the correct way to get these classes?

**Comment by AM-NiceDay** [comment, marked as answer]
Hey @vaelu!

The right key here is `family-name`:
```html
<h1 class="font-[family-name:var(--headline1-font-family)]">Title</h1>
```

Here you can find full example: https://play.tailwindcss.com/yzo0HmrR9u.

**Reply by vaelu** [reply]
Works, many thanks! 🚀

I think this would be a nice addition for the docs, since I cannot find this `family-name` documented anywhere?

**Reply by Darkdecus** [reply]
slight correction as at the time I'm seeing this, you will have to remove the `var` and the `()` else the style won't work, so the right class should be
```html
<h1 class="font-[family-name:--headline1-font-family]">Title</h1>
```
